cpp/downgrades/9a7c3c14c1076f64b871719117a558733d987b48/derivedtypes.ql

```diff
@@ -0,0 +1,19 @@
+class Type extends @type {
+  string toString() { none() }
+}
+
+predicate derivedType(Type type, string name, int kind, Type type_id) {
+  derivedtypes(type, name, kind, type_id)
+}
+
+predicate typeTransformation(Type type, string name, int kind, Type type_id) {
+  type_operators(type, _, _, type_id) and
+  name = "" and
```

Comment:
I'm guessing this `name = ""` means losing some information, but we're accepting that, right?

Reply:
Not really. I'm just projecting all the rows of the `type_operators` table on the `derivedtypes` table, to make sure I have some kind of type, but the `derivedtypes` table requires me to make up some kind of name.

Reply:
I could compute something based on the two columns of `type_operators` that I ignore here, but that doesn't seem to be worth the trouble.